app/views/Project/Tagging/Sources/SourcesTable/LeadCopyModal/index.tsx

```diff
@@ -0,0 +1,242 @@
+                id
+                title
+            }
+        }
+    }
+`;
+
+const LEAD_COPY = gql`
+    mutation LeadCopy(
+        $activeProjectId: ID!,
+        $projectIds: [ID!]!,
+        $leadIds: [ID!]!,
+    ) {
+        project(id: $activeProjectId) {
+            leadCopy(data: {
+                projects: $projectIds,
+                leads: $leadIds,
+            }) {
+                errors
+                ok
+            }
+        }
+    }
+`;
+
+type ProjectItem = NonNullable<NonNullable<NonNullable<ProjectListForLeadCopyQuery['projects']>['results']>[number]>;
+
+const searchProjectsKeySelector = (d: ProjectItem) => d.id;
+const searchProjectsLabelSelector = (d: ProjectItem) => d.title;
+const selectedProjectsKeySelector = (d: ProjectItem) => d.id;
+
+interface SelectedProjectsProps {
+    title: string;
+}
+function SelectedProjectsRenderer(props: SelectedProjectsProps) {
+    const {
+        title,
+    } = props;
+    return (
+        <>
+            {title}
+        </>
+    );
+}
+
+interface Props {
+    onClose: () => void;
+    leadId: string[];
+    projectId: string;
+}
+
+function LeadCopyModal(props: Props) {
+    const {
+        onClose,
+        projectId,
+        leadId,
+    } = props;
+
+    const [opened, setOpened] = useState(false);
+    const [searchText, setSearchText] = useState<string>('');
+    const debouncedSearchText = useDebouncedValue(searchText);
+    const alert = useAlert();
+
+    const searchProjectsVariables = useMemo(() => ({
+        search: debouncedSearchText,
+    }), [debouncedSearchText]);
+
+    const [selectedProjects, setSelectedProjects] = useState<string[]>([]);
+    const [projectOptions, setProjectOptions] = useState<ProjectItem[] | undefined | null>([]);
+
+    const {
+        loading,
+        data,
+    } = useQuery<ProjectListForLeadCopyQuery, ProjectListForLeadCopyQueryVariables>(
+        PROJECT_LIST_FOR_LEAD_COPY,
+        {
+            variables: searchProjectsVariables,
+            skip: !opened,
+            onCompleted: (response) => {
+                const userProjects = response?.projects?.results;
+                const filteredProjects = userProjects?.filter(
+                    (u: ProjectItem) => u.id !== projectId,
+                );
+                setProjectOptions(filteredProjects);
+            },
+        },
+    );
+
+    const [
+        leadCopy,
+    ] = useMutation<LeadCopyMutation, LeadCopyMutationVariables>(
+        LEAD_COPY,
+        {
+            onCompleted: () => {
+                alert.show(
+                    'Successfully copied leads to projects',
+                    {
+                        variant: 'success',
+                    },
+                );
+            },
+            onError: () => {
+                alert.show(
+                    'Failure copying leads',
+                    {
+                        variant: 'error',
+                    },
+                );
+            },
```

Comment:
Let's add better success/error messages

onCompleted does not guarantee success, should check 'ok' status